src/ProblemDetails/ProblemDetailsMiddleware.cs

```
@@ -85,6 +85,11 @@ public async Task Invoke(HttpContext context)
 
                     var details = GetDetails(context, error);
 
+                    if (details == null) // Re-throw the original exception if we can't or don't want to map it to ProblemDetails
+                    {
+                        throw;
```

Re-throwing here will remove the option of logging the exception, is that intended? It will also trigger the `Logger.ProblemDetailsMiddlewareException` call below, which I don't think we want.

---

good point. I think we shouldn't log anything if we're wilfully ignoring the exception. Or maybe we may just log that we're wilfully ignoring that exception.

---

@khellang what do you think - should we ignore the exception completely, or should we log that an exception has occured, but custom mapping rules decided to ignore it?

---

I think we should at least give the user the options to choose (by allowing them to turn it off using `ShouldLogUnhandledException`)

---

`ShouldLogUnhandledException` expects `MvcProblemDetails` as input, and we're dealing with a case when `MvcProblemDetails` is deliberately `null`, so I opted for logging always, but on `Information` level.